sogs/model/room.py

```diff
@@ -1151,7 +764,7 @@ def delete_posts(self, message_ids: List[int], deleter: User):
                 )
 
                 if ids:
-                    if not self.check_moderator(deleter):
+                    if not self.default_bot.check_moderator(self, deleter):
```

Comment:
These `default_bot.check_moderator` and other similar functions feel wrong. 

The call should be something more like `user.is_moderator` and have an authentication module in the root level directory that fills out this when the request is received

Reply:
is there a reason that room model can't have the `check_moderator` method that just calls `default_bot.check_moderator` ?

Reply:
(a lot less would have been modified that way)

Reply:
> oh god, this is the last time i use VS code.

Fixed that for you, I hope.

Reply:
all removed